invenio_vocabularies/services/facets.py

```diff
@@ -53,17 +54,22 @@ def _vocab_to_label(self, vocab):
     def __call__(self, ids):
         """Return the mapping when evaluated."""
         identity = AnonymousIdentity()
-        if not self.cache:
-            vocabs = self.service.read_many(
-                identity, type=self.vocabulary, ids=ids, fields=self.fields
-            )
-        else:
-            vocabs = self.service.read_all(
-                identity, type=self.vocabulary, fields=self.fields
-            )
+        try:
+            if not self.cache:
+                vocabs = self.service.read_many(
+                    identity, type=self.vocabulary, ids=ids, fields=self.fields
+                )
+            else:
+                vocabs = self.service.read_all(
+                    identity, type=self.vocabulary, fields=self.fields
+                )
+            vocab_list = list(vocabs.hits)  # the service returns a generator
+        except NoResultFound:
+            # in the case the vocabulary type does not exist
+            # facets should not fail but be empty
+            vocab_list = []
```

Comment:
**alternative**: create a custom exception, that then would be raised in the except and captured as a 500.

Reply:
Question: this is called if the facet is defined, and it should work...
If I don't define a facet on this vocabulary, then it should not be called (so it should not fail).

If I configure the facet, I would probably prefer to have a clear exception. Instead, I see an empty facet and I won't understand why?
Let me know if I have maybe misunderstood this...

Reply:
What you are saying is correct. I still think a full blown 500 is an overkill for something that has a small-medium impact on the user. But since we are not using logging consistently (i.e. to alert sysadmins), I will create a new error (e.g. `FacetNotFound` and raise it)

Reply:
Implemented as discussed in Discord